packages/web/pages/wormhole.tsx

```diff
@@ -176,24 +354,57 @@ const Wormhole: FunctionComponent = () => {
   };
 
   let bridgeDefaults = {
-    fromNetwork: "solana",
-    toNetwork: "osmosis",
+    fromNetwork: "solana" as ChainName,
+    toNetwork: "osmosis" as ChainName,
     token: "W",
-    requiredNetwork: "osmosis",
+    requiredNetwork: "osmosis" as ChainName,
   };
 
   if (fromNetwork) {
-    bridgeDefaults.fromNetwork = fromNetwork;
+    bridgeDefaults.fromNetwork = fromNetwork as ChainName;
   }
   if (toNetwork) {
-    bridgeDefaults.toNetwork = toNetwork;
+    bridgeDefaults.toNetwork = toNetwork as ChainName;
   }
   if (token) {
     bridgeDefaults.token = token;
   }
   config.bridgeDefaults = bridgeDefaults;
 
-  return <WormholeConnect config={config} theme={customTheme} />;
+  useEffect(() => {
+    const script = document.createElement("script");
+    script.type = "module";
+    script.src =
+      "https://www.unpkg.com/@wormhole-foundation/wormhole-connect@0.3.16/dist/main.js";
+    script.defer = true;
+    script.onload = () => setScriptLoaded(true);
+    document.body.appendChild(script);
+
+    return () => {
+      document.body.removeChild(script);
+    };
+  }, []);
+
+  return (
+    <>
+      {!scriptLoaded && (
+        <div className="flex h-screen w-full items-center justify-center">
+          <Spinner />
+        </div>
+      )}
+      <div
+        id="wormhole-connect"
+        data-config={JSON.stringify(config)}
+        data-theme={JSON.stringify(customTheme)}
+        style={{ display: scriptLoaded ? "block" : "none" }}
+      ></div>
+
+      <link
+        rel="stylesheet"
```

Comment:
Just for security reason can we use the [`integrity`](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) attribute so we make sure the file hash is valid?

Reply:
Good catch! I'll open a PR adding it